lang/java/BUILD.md

```diff
@@ -44,6 +46,12 @@ To perform "fast" build, which skips tests and all code quality enforcement tool
 
     mvn clean install -DskipTests -TC1 -P!code-quality
 
+
+  * Please follow the following wiki for all running all java builds [Java Build Wiki](https://cwiki.apache.org/confluence/display/REEF/Linux), the REEF unit tests require a number of open files which is greater than the default open file limit on a number of Linux distributions such as Ubuntu 16.04/16.10.  This limit is controlled in the shell by the "ulimit -n" command
```

Comment:
This is problematic for two reasons: It dates the documentation to a specific Ubuntu release and it refers back to the wiki we aim to replace with these concise build instructions. How about the following instead?

```
*Note:* The integration tests open a large number of files in parallel. Depending on your system configuration, this can lead to a spurious test failure. On Linux, you might have to use the `ulimit -n` command to adjust the limit on the number of open files allowed.
```

It might also make sense to put this paragraph further above, as someone who reads from the top and executes the first command we suggest will already have had the problem by the time they reach this. Maybe add it to the `Prerequisites`?